arangod/RestServer/Metrics.h

```diff
@@ -93,52 +93,74 @@ template<typename T> class Gauge : public Metric {
   Gauge(T const& val, std::string const& name, std::string const& help,
         std::string const& labels = std::string())
     : Metric(name, help, labels) {
-    _g.store(val);
+    _g.store(val, std::memory_order_relaxed);
   }
   Gauge(Gauge const&) = delete;
   ~Gauge() = default;
   std::ostream& print (std::ostream&) const;
-  Gauge<T>& operator+=(T const& t) {
+
+  T fetch_add(T t, std::memory_order mo = std::memory_order_relaxed) noexcept {
```

Comment:
Not sure if we should use `memory_order_relaxed` as default here. I don't think we will ever have a case where we actually need something stronger than relaxed, but it feels not right. However, if we use `memory_order_seq_cst` as default order we should probably adapt all callers (even though on x86 we wouldn't gain anything), which would be a much larger change. So guess for now we can leave it as is.

Reply:
i did see this dilemma and also decided that this was the pragmatic solution